### PR TITLE
Add unit tests for configs.Hooks

### PR DESCRIPTION
**Pull request by albertoleal**
This PR only adds unit tests to config.Hooks.

### CODE REVIEW

**Comment by albertoleal** [comment]
Any feedback on this PR? cc @LK4D4

File: libcontainer/configs/config.go [hunk 1]
```diff
@@ -228,6 +228,8 @@ func (hooks *Hooks) UnmarshalJSON(b []byte) error {
 
 func (hooks Hooks) MarshalJSON() ([]byte, error) {
 	serialize := func(hooks []Hook) (serializableHooks []CommandHook) {
+		serializableHooks = []CommandHook{}
```
Comment (cyphar) [inline libcontainer/configs/config.go]: Why is this necessary? Is it because you need to get `[]`? In that case, can you remove the named return value?

**Comment by albertoleal** [comment]
Hey @cyphar, just removed the named parameter.

**Comment by cyphar** [comment]
You don't appear to test that `Marshal -> Unmarshal` works as expected. But apart from that, LGTM.

**Comment by albertoleal** [comment]
You mean having one tests checking for `Prestart`,`Poststart`and `Poststop`? If so, I can change it..

**Comment by cyphar** [comment]
What I meant was that you test `hook.Unmarshal(hook.Marshal)` to make sure that the result is the original hook. Your marshal and unmarshal tests either use `Sprintf` or compare to hard-coded values.

**Comment by albertoleal** [comment]
Got it 👍 
Will create another scenario for that and update this PR.

**Comment by albertoleal** [comment]
@cyphar Just added one more scenario using both methods (`Marshal` and `Unmarshal`). 

One thing that I was tempted to change is the way we're returning the hooks and they're empty. Currently we're returning `null`:
```
{"poststart":null,"poststop":null,"prestart":null}
```
 instead of an empty list `[]`:
```
`{"poststart":[],"poststop":[],"prestart":[]}`
```

I decided to keep the existing behaviour on this PR. Let me know if you think it's worth to apply this change, maybe I can submit another PR since it may break something else.

**Comment by crosbymichael** [comment]
LGTM

**Comment by cyphar** [comment]
LGTM. Merging.
